doc/tutorials/acrn-secure-boot-with-grub.rst

```diff
@@ -0,0 +1,177 @@
+.. _how-to-enable-acrn-secure-boot-with-grub:
+
+Enable ACRN Secure Boot with GRUB
+#################################
+
+This document shows how to enable ACRN secure boot with GRUB including:
+
+  -  ACRN Secure Boot Sequence
+  -  Generate GPG Key
+  -  Setup Standalone GRUB EFI Binary
+  -  UEFI Secure Boot
+
+  Validation Environment:
+	Hardware Platform: INTEL WHL
+	ACRN Scenario: Industry
+	Service VM: Ubuntu 18.04
+	GRUB: 2.02
+
+Note: Please make sure you have alternative ways to restore GRUB when you got stuck.
+
+ACRN Secure Boot Sequence
+*************************
+
+  ACRN can be booted by Multiboot compatible bootloader, following diagram illustrates the boot sequence of ACRN with GRUB:
+
+     .. image:: images/acrn-secureboot-flow.png
+        :align: center
+        :width: 600px
+
+
+  Refer to :ref:`using_grub` for details to enable GRUB on ACRN.
+
+  From point view of secureboot:
+	- UEFI firmware verifies Shim/GRUB
+	- GRUB verifies ACRN, Service VM kernel and Prelaunch VM kernel
+	- SOS kernel verifies DM and UOS OVMF thru ACRN DM
+	- UOS OVMF starts the guest side verified boot process
+
+  This document provides guide to enable verification function of GRUB to verify ACRN binaries(such acrn.bin, service VM bzImage and Prelaunch VM kernel image if it is present).
+
+
+Generate GPG Key
+****************
+
+  GRUB supports loading of GPG signed files only if digital signatures enabled.
+  Example to generate GPG signing key::
+  $mkdir --mode 0700 keys
+  $gpg --homedir keys  --gen-key
+  $gpg --homedir keys --export > boot.key
+
+  ``gpg --gen-key`` generates key pair, private key will be used to sign GRUB configuration files and ACRN binaries, public key will be embedded in GRUB and is used to verify GRUB configuration files or binaries GRUB trys to load.
+
+
+Setup Standalone GRUB EFI Binary
+********************************
+
+Prepare Initial GRUB Configuration grub.init.cfg
+================================================
+  Store the following minimal grub config in grub.init.cfg, ``set check_signatures =enforce`` is to enable digital signatures in GRUB::
+	set check_signatures=enforce
+	export check_signatures
+
+	search --no-floppy --fs-uuid --set=root ESP_UUID
+	configfile /grub.cfg
+	echo /grub.cfg did not boot the system, rebooting in 10 seconds.
+	sleep 10
+	reboot
+
+  Replace the ESP_UUID with UUID of your EFI system partition, with the following output of ``lsblk -f``, replace it with ``24FC-BE7A``.::
```

Comment:
@dbkinder can you help reword this sentence?

Also, just wondered where we should replace the double-backticks around `lbblk` by :command:`lsblk -f`?